zarr/tests/test_core.py

```diff
@@ -194,37 +213,84 @@ def test_array_2d(self):
         eq(a.nbytes, z.nbytes)
         eq(50, z.nchunks_initialized)
 
-        # check slicing
+        # check array-like
         assert_array_equal(a, np.array(z))
+
+        # check slicing
+
+        # total slice
         assert_array_equal(a, z[:])
         assert_array_equal(a, z[...])
         # noinspection PyTypeChecker
         assert_array_equal(a, z[slice(None)])
+
+        # slice first dimension
         assert_array_equal(a[:10], z[:10])
         assert_array_equal(a[10:20], z[10:20])
         assert_array_equal(a[-10:], z[-10:])
+        assert_array_equal(a[:10, :], z[:10, :])
+        assert_array_equal(a[10:20, :], z[10:20, :])
+        assert_array_equal(a[-10:, :], z[-10:, :])
+        assert_array_equal(a[:10, ...], z[:10, ...])
+        assert_array_equal(a[10:20, ...], z[10:20, ...])
+        assert_array_equal(a[-10:, ...], z[-10:, ...])
+        assert_array_equal(a[:10, :, ...], z[:10, :, ...])
+        assert_array_equal(a[10:20, :, ...], z[10:20, :, ...])
+        assert_array_equal(a[-10:, :, ...], z[-10:, :, ...])
+
+        # slice second dimension
         assert_array_equal(a[:, :2], z[:, :2])
         assert_array_equal(a[:, 2:4], z[:, 2:4])
         assert_array_equal(a[:, -2:], z[:, -2:])
+        assert_array_equal(a[..., :2], z[..., :2])
+        assert_array_equal(a[..., 2:4], z[..., 2:4])
+        assert_array_equal(a[..., -2:], z[..., -2:])
+        assert_array_equal(a[:, ..., :2], z[:, ..., :2])
+        assert_array_equal(a[:, ..., 2:4], z[:, ..., 2:4])
+        assert_array_equal(a[:, ..., -2:], z[:, ..., -2:])
+
+        # slice both dimensions
         assert_array_equal(a[:10, :2], z[:10, :2])
         assert_array_equal(a[10:20, 2:4], z[10:20, 2:4])
         assert_array_equal(a[-10:, -2:], z[-10:, -2:])
-        # ...across chunk boundaries...
+
+        # slicing across chunk boundaries
         assert_array_equal(a[:110], z[:110])
         assert_array_equal(a[190:310], z[190:310])
         assert_array_equal(a[-110:], z[-110:])
+        assert_array_equal(a[:110, :], z[:110, :])
+        assert_array_equal(a[190:310, :], z[190:310, :])
+        assert_array_equal(a[-110:, :], z[-110:, :])
         assert_array_equal(a[:, :3], z[:, :3])
         assert_array_equal(a[:, 3:7], z[:, 3:7])
         assert_array_equal(a[:, -3:], z[:, -3:])
         assert_array_equal(a[:110, :3], z[:110, :3])
         assert_array_equal(a[190:310, 3:7], z[190:310, 3:7])
         assert_array_equal(a[-110:, -3:], z[-110:, -3:])
-        # single item
+
+        # single row/col/item
         assert_array_equal(a[0], z[0])
         assert_array_equal(a[-1], z[-1])
+        assert_array_equal(a[:, 0], z[:, 0])
+        assert_array_equal(a[:, -1], z[:, -1])
         eq(a[0, 0], z[0, 0])
         eq(a[-1, -1], z[-1, -1])
 
```

Comment:
I would strongly recommend adding some short-form cases for vectorized indexing (i.e., with `.vindex`). You have partial test coverage for this already, but there are some many indexing edge cases that it's a good idea to write them in the most succinct way possible.

Reply:
I've added some tests to cover these cases. Still a bit more coverage needed.